packages/hierarchy-builder/src/hierarchy-builder/HierarchyNode.ts

```diff
@@ -228,6 +228,101 @@ export namespace HierarchyNodeKey {
       }
     }
   }
+  /**
+   * Compares two given keys.
+   * @returns
+   *- `0` if they are equal
+   *- `negative value` if lhs key is less than rhs key
+   *- `positive value` if lhs key is more than rhs key
+   */
+  export function compare(lhs: HierarchyNodeKey, rhs: HierarchyNodeKey): number {
+    // This compare is quicker than String.localeCompare
+    function compareStrings(lhsStr: string, rhsStr: string): number {
+      return lhsStr < rhsStr ? -1 : lhsStr > rhsStr ? 1 : 0;
+    }
+    if (isCustom(lhs)) {
+      if (!isCustom(rhs)) {
+        return 1;
+      }
+      return compareStrings(lhs, rhs);
+    }
+    if (isCustom(rhs)) {
+      return -1;
+    }
+
+    const typeCompareResult = compareStrings(lhs.type, rhs.type);
+    if (typeCompareResult !== 0) {
+      return typeCompareResult;
+    }
+
+    switch (lhs.type) {
+      case "instances": {
+        assert(isInstances(rhs));
+        if (lhs.instanceKeys.length !== rhs.instanceKeys.length) {
+          return lhs.instanceKeys.length > rhs.instanceKeys.length ? 1 : -1;
+        }
+        for (let i = 0; i < lhs.instanceKeys.length; ++i) {
+          const lhsInstanceKey = lhs.instanceKeys[i];
+          const rhsInstanceKey = rhs.instanceKeys[i];
+          const classNameComparResult = compareStrings(lhsInstanceKey.className, rhsInstanceKey.className);
+          if (classNameComparResult !== 0) {
+            return classNameComparResult;
+          }
+          const idCompareResult = compareStrings(lhsInstanceKey.id, rhsInstanceKey.id);
+          if (idCompareResult !== 0) {
+            return idCompareResult;
+          }
```

Comment:
Let's move this to [InstanceKey namespace](https://github.com/iTwin/presentation/blob/53893a975dce944a356ff7597103466c0fa8ff25/packages/hierarchy-builder/src/hierarchy-builder/values/Values.ts#L28) as a `compare` function.

Reply:
Moved